src/main/java/BoardCopier.java

```diff
@@ -9,8 +7,54 @@ public BoardCopier() {}
      * @param board the board to be cloned
      * @return a clone of the board
      */
-    public Board createCopy(Board board) {
-        // TODO
-        return null;
+
+    private Board newBoard = new Board();
+    private Board oldBoard;
+
+    public BoardCopier(Board board) {
+        this.oldBoard = board;
+    }
+
+    public Board createCopy() {
+        Player[] oldPlayers = this.oldBoard.getPlayers();
+        clonePlayer(oldPlayers);
+        clonePieces();
+        //TODO!!!
+        // Don't entirely understand the use of mirror coordinates
+        // If still needed, will need to implement later
+
+        return this.newBoard;
+    }
+
+    private void clonePieces() {
+        for (int i=0; i<Board.HEIGHT; i++) {
+            for (int j=0; j<Board.WIDTH; j++) {
```

Comment:
Please annotate the purpose of absPiece, relPiece, mirrorPiece. At a glance it seems redundant to create clone pieces for at least absPiece and mirrorPiece (ignoring relPiece) because all coordinates are already being looped through. This is because, as far as I understand it, these three are just different perspectives on what is ultimately the same board, and so each piece would be cloned three times.